range.go

```diff
@@ -217,6 +243,144 @@ func splitComparatorVersion(s string) (string, string, error) {
 	return strings.TrimSpace(s[0:i]), s[i:], nil
 }
 
+// getWildcardType will return the type of wildcard that the
+// passed version contains
+func getWildcardType(vStr string) wildcardType {
+	parts := strings.Split(vStr, ".")
```

Comment:
That's only an issue if pr and build metadata versions are allowed as ranges, sadly i haven't written tests for that case in the first place, therefor that's open for discussion